core/station/impl/src/services/asset.rs

```diff
@@ -0,0 +1,303 @@
+use std::sync::Arc;
+
+use crate::{
+    core::{
+        authorization::Authorization, generate_uuid_v4, utils::retain_accessible_resources,
+        CallContext,
+    },
+    errors::AssetError,
+    models::{
+        resource::{Resource, ResourceAction, ResourceId},
+        AddAssetOperationInput, Asset, AssetCallerPrivileges, AssetId, EditAssetOperationInput,
+        RemoveAssetOperationInput,
+    },
+    repositories::{AssetRepository, ASSET_REPOSITORY},
+};
+use lazy_static::lazy_static;
+use orbit_essentials::{
+    api::ServiceResult,
+    model::ModelValidator,
+    pagination::{paginated_items, PaginatedData, PaginatedItemsArgs},
+    repository::Repository,
+};
+use station_api::ListAssetsInput;
+use uuid::Uuid;
+
+lazy_static! {
+    pub static ref ASSET_SERVICE: Arc<AssetService> =
+        Arc::new(AssetService::new(Arc::clone(&ASSET_REPOSITORY),));
+}
+
+#[derive(Default, Debug)]
+pub struct AssetService {
+    asset_repository: Arc<AssetRepository>,
+}
+
+impl AssetService {
+    pub const DEFAULT_LIST_ASSETS_LIMIT: u16 = 100;
+    pub const MAX_LIST_ASSETS_LIMIT: u16 = 1000;
+
+    pub fn new(asset_repository: Arc<AssetRepository>) -> Self {
+        Self { asset_repository }
+    }
+
+    pub fn get(&self, asset_id: &AssetId) -> ServiceResult<Asset> {
+        let asset = self
+            .asset_repository
+            .get(asset_id)
+            .ok_or(AssetError::NotFound {
+                id: Uuid::from_bytes(*asset_id).hyphenated().to_string(),
+            })?;
+
+        Ok(asset)
+    }
+
+    pub async fn create(&self, input: AddAssetOperationInput) -> ServiceResult<Asset> {
+        let id = generate_uuid_v4().await;
```

Comment:
now that we have the `Uuid::new_v4` being seeded you could potentially use that here and remove the async need of this method